app/lib/file_parse_service.rb

```diff
@@ -76,11 +80,12 @@ def self.run_parse_job(study_file, study, user, reparse: false, persist_on_fail:
         end
       when '10X Barcodes File'
         # push immediately to avoid race condition when initiating parse
-        study.delay.send_to_firecloud(study_file)
-        bundle = study_file.study_file_bundle
-        matrix = bundle.parent
-        genes = bundle.bundled_files.detect {|f| f.file_type == '10X Genes File' }
-        if genes.present? && matrix.present? && bundle.completed?
+        study.delay.send_to_firecloud(study_file) if study_file.is_local?
+        study_file.reload
+        if study_file.study_file_bundle.try(:completed?)
+          bundle = study_file.study_file_bundle
+          matrix = bundle.parent
+          genes = bundle.bundled_files.detect {|f| f.file_type == '10X Genes File' }
```

Comment:
```suggestion
          sibling_file_type = study_file.file_type == '10X Genes File' ? '10X Barcodes File' : '10X Genes File'
          sibling_file = bundle.bundled_files.detect {|f| f.file_type == sibling_file_type }
```
If you do that, I believe you can consolidate the Genes and Barcodes logic into a single `when`.  (and with something a bit more complex, you could probably also consolidate the Coordinate Matrix `when`, although that might not be worth it)

Reply:
Nice idea - I'll give that a shot.

Reply:
b46fb50